ocs_ci/ocs/pgsql.py

```diff
@@ -663,3 +664,193 @@ def pgsql_full(self):
+            postgres_pod (POD) : POD object on which postgres is running.
+            max_row_index (int) : Maximum index of the row currently present in the table.
+            table_name (str) : Name of the table.
+
+        """
+        log.info(f"Running {num_of_ops} update operations.")
+        self.run_pgsql_command(
+            postgres_pod,
+            (
+                f"UPDATE {table_name} SET username='{self.generate_random_string(50)}' "
+                f"WHERE row_id in {tuple(random.sample(range(1, max_row_index), num_of_ops))};"
+            ),
+        )
+
+    def run_delete_operation(self, num_of_ops, postgres_pod, max_row_index, table_name):
+        """
+        Delete num_of_ops number of rows from the table.
+
+        Args:
+            num_of_ops (int) : Total number of records we want to be deleted.
+            postgres_pod (POD) : POD object on which postgres is running.
+            max_row_index (int) : Maximum index of the row currently present in the table.
+            table_name (str) : Name of the table.
+
+        """
+        log.info(f"Running {num_of_ops} delete operations.")
+        self.run_pgsql_command(
+            postgres_pod,
+            f"DELETE FROM {table_name} WHERE row_id in {tuple(random.sample(range(1, max_row_index), num_of_ops))};",
+        )
+
+    def run_pgsql_queries(
+        self,
+        postgres_pod,
+        table_name,
+        total_rows,
+        num_of_ops,
+        run_time,
+        drop_table=True,
+    ):
+        """
+        Run queries on the Postgres database,
+        The steps include:
+        1. Create table <table_name> and insert <total_rows> number of records.
+        2. Repeat below steps in order for specified <run_time> duration:
+        a. Insert <num_of_ops> number of rows into the table.
+        b. Update <num_of_ops> number of rows into the table.
+        c. Update random number of rows based on matching year in the date column of records.
+        d. Delete <num_of_ops> number of rows from the table.
+
+        Args:
+            postgres_pod (POD) : POD object on which postgres is running.
+            table_name (str) : Name of the table that we want to create.
+            total_rows (int) : Total number of records that should be there inside the table.
+            num_of_ops (int) : Total number of records we want to be inserted/deleted/updated in each operation
+                                respectively.
+            run_time (int) : Total Run Time in minutes.
+            drop_table (bool) : True if we want to delete the table, False otherwise.
+
+        Returns:
+             str  : A string containing the last valid state of the table.
+
+        """
+        row_index = 0
+        last_valid_state = ""
+        self.run_pgsql_command(
+            postgres_pod,
+            f"CREATE TABLE {table_name} (row_id INT PRIMARY KEY, username VARCHAR (50) NOT NULL, date DATE NOT NULL);",
```

Comment:
Can you make it more generic, so the table structure will be take from an outside file (YAML / JSON) ?

this is relevant for all other functions where the table filed are constants.

Reply:
The table and its data cannot be generated running a YAML, we can read the content from the YAML/JSON and based on that we can construct the data to be inserted into the table but not sure if that is really needed for this use case and I do not see much benefit implementing that way.